sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -282,14 +239,17 @@ private[parquet] class ParquetRowConverter(
       // Create a RowUpdater instance for converting Parquet objects to Catalyst rows. If any fields
       // in the Catalyst result schema have associated existence default values, maintain a boolean
       // array to track which fields have been explicitly assigned for each row.
-      val rowUpdater: RowUpdater =
-        if (catalystType.hasExistenceDefaultValues) {
-          resetExistenceDefaultsBitmask(catalystType)
-          new RowUpdaterWithBitmask(
-            currentRow, catalystFieldIndex, catalystType.existenceDefaultsBitmask)
-        } else {
-          new RowUpdater(currentRow, catalystFieldIndex)
+      val rowUpdater: RowUpdater = new RowUpdater(currentRow, catalystFieldIndex)
+      if (catalystType.hasExistenceDefaultValues) {
+        for (i <- 0 until catalystType.existenceDefaultValues.size) {
+          catalystType.existenceDefaultsBitmask(i) =
+            if (i < parquetType.getFieldCount) {
```

Comment:
Why comparing with `parquetType.getFieldCount` here?

Reply:
We discussed this offline. I moved this code out of the `parquetType.getFields.asScala.map { parquetField => ...` loop, and also ported the explanation into a comment here:

```
// Assume the schema for a Parquet file-based table contains N fields. Then if we later
// run a command "ALTER TABLE t ADD COLUMN c DEFAULT <value>" on the Parquet table, this
// adds one field to the Catalyst schema. Then if we query the old files with the new
// Catalyst schema, we should only apply the existence default value to all columns > N.
```